### PR TITLE
adding support for multiple sub-chapters

### LINKED ISSUE #14: Should the TOC links point to /notebooks or to /ch
Currently the TOC links point to /ch, which is where the converted notebooks will be placed. However, this is a bit unintuitive as it requires the person understand *where* the pages will go, not where they are now.

The old version of this repo had the TOC links go to /notebooks, and this was auto-converted to /ch at build. Is that a better solution? Otherwise, we need to document the /ch behavior

### PR DESCRIPTION
**Pull request by choldgraf**
This adds support for multiple chapters in the base URL. It accomplishes this by using `_build` as a jekyll collection for the built content, and then a permalink rule for this collection that simply drops the collection name from the URL.

The result is that a notebook structure like

```
content/ch1/ntbk.ipynb
content/ch2/ntbk2.ipynb
```

will have permalinks

```
mysite.org/ch1/ntbk/
mysite.org/ch2/ntbk2/
```

Could be used for something like #20 but closes #14 

## Changes to the build

* TOC links should now be *relative to the `/content` directory*
* All site files must now be in the `/content` directory and specified in the `toc.yml` file to show up

### CODE REVIEW

File: _includes/sidebar.html [hunk 1]
```diff
@@ -20,19 +20,21 @@
...
+      {% elsif chapter.header %}
+        <li><h2 class="c-sidebar__title">{{ chapter.header }}</li>
+        {% continue %}
       {% endif %}
 
       {% comment %}
       If the entry is the current page, assign a CSS class to highlight it.
       {% endcomment %}
-      {% if chapter.url == page.url %}
+      {% if page.url contains chapter.url %}
```
Comment (choldgraf) [inline _includes/sidebar.html:37]: this is changed because the page url now ends with a `/` while the chapter URL does not (because it points to a file)

**Comment by choldgraf** [comment]
note to @SamLau95 and @TomDonoghue since they both showed interest in this (most of the files in this PR are just changes from the old->new build system)

**Comment by SamLau95** [comment]
This looks good to me!

**Comment by choldgraf** [comment]
@SamLau95 you should merge if you're +1 :-)

**Comment by TomDonoghue** [comment]
Just chiming in late - this looks awesome! We've been playing with the new version of the textbook, and the multiple sections really helps with our set up (and fixes the current silly hack of hosting three different textbooks, that are linked together).

**Pull request merged**